two errors:
```
________ sympy/geometry/tests/test_ellipse.py:test_object_from_equation ________
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/utilities/runtests.py", line 1306, in _timeout
    function()
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/geometry/tests/test_ellipse.py", line 35, in test_object_from_equation
    assert Circle(x**2 + y**2 + 3*x + 4*y + 26) == Circle(Point2D(-S(3)/2, -2), sqrt(79)*I/2)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/geometry/ellipse.py", line 1426, in __new__
    return Circle((center_x, center_y), sqrt(r2), evaluate=evaluate)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/geometry/ellipse.py", line 1440, in __new__
    r = Point(args[1], 0, evaluate=evaluate).x  # convert via Point as necessary
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/geometry/point.py", line 158, in __new__
    raise ValueError('Imaginary coordinates are not permitted.')
ValueError: Imaginary coordinates are not permitted.
________________________________________________________________________________
____________ sympy/geometry/tests/test_ellipse.py:test_issue_15797 _____________
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/utilities/runtests.py", line 1306, in _timeout
    function()
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/geometry/tests/test_ellipse.py", line 477, in test_issue_15797
    assert u.center.x.is_Float
AssertionError
```

The first could use `Circle(Circle((0, 1), 2).equation()) == Circle((0, 1), 2)`. I am not sure what is wrong with the second one.

--- a/sympy/geometry/tests/test_ellipse.py
+++ b/sympy/geometry/tests/test_ellipse.py
@@ -471,7 +471,9 @@ def test_issue_15259():
 
 
 def test_auxiliary_circle():
-    x, y, a, b, e = symbols('x y a b e')
+    x, y, a, b = symbols('x y a b')
     e = Ellipse((x, y), a, b)
+    # the general result
     assert e.auxiliary_circle() == Circle((x, y), Max(a, b))
+    # a special case where Ellipse is a Circle
     assert Circle((3, 4), 8).auxiliary_circle() == Circle((3, 4), 8)

--- a/sympy/geometry/tests/test_ellipse.py
+++ b/sympy/geometry/tests/test_ellipse.py
@@ -436,6 +436,7 @@ def test_parameter_value():
     assert e.parameter_value((3, 0), t) == {t: 0}
     raises(ValueError, lambda: e.parameter_value((4, 0), t))
 
+
 def test_second_moment_of_area():
     x, y = symbols('x, y')
     e = Ellipse(Point(0, 0), 5, 4)
@@ -446,6 +447,7 @@ def test_second_moment_of_area():
     assert I_yy == e.second_moment_of_area()[1]
     assert I_xx == e.second_moment_of_area()[0]
     assert I_xy == e.second_moment_of_area()[2]
+
 
 def test_circumference():
     M = Symbol('M')
@@ -462,48 +464,14 @@ def test_circumference():
 
     # test numerically
     assert abs(Ellipse(None, hradius=5, vradius=3).circumference.evalf(16) - 25.52699886339813) < 1e-10
+
+
 def test_issue_15259():
     assert Circle((1, 2), 0) == Point(1, 2)
 
 
 def test_auxiliary_circle():
-    x, y = symbols('x, y')
-
-    p1 = Point(0, 0)
-    p2 = Point(3, 4)
-
-    e1 = Ellipse(p1, 1,1)
-    e2 = Ellipse(p2, 2,2)
-
-    c1 = Circle(p1, 4)
-    c2 = Circle(p2, 8)
-
-    e3 = Ellipse(p1, 5,2)
-    e4 = Ellipse(p2, 6,4)
-
-    e5 = Ellipse(p1, 4,7)
-    e6 = Ellipse(p2, 5,9)
-
-    # circle If self.hradius = self.vradius
-
-    assert e1.auxiliary_circle() == Circle(Point2D(0, 0), 1)
-    assert e2.auxiliary_circle() == Circle(Point2D(3, 4), 2)
-    assert c1.auxiliary_circle() == Circle(Point2D(0, 0), 4)
-    assert c2.auxiliary_circle() == Circle(Point2D(3, 4), 8)
-
-    # horizontal ellipse If self.hradius > self.vradius
-
-    assert e3.auxiliary_circle() == Circle(Point2D(0, 0), 5)
-    assert e4.auxiliary_circle() == Circle(Point2D(3, 4), 6)
-
-    # vertical ellipse If self.hradius < self.vradius
-
-    assert e5.auxiliary_circle() == Circle(Point2D(0, 0), 7)
-    assert e6.auxiliary_circle() == Circle(Point2D(3, 4), 9)
-
-
-    def test_auxiliary_circle():
     x, y, a, b, e = symbols('x y a b e')
     e = Ellipse((x, y), a, b)
     assert e.auxiliary_circle() == Circle((x, y), Max(a, b))
-
+    assert Circle((3, 4), 8).auxiliary_circle() == Circle((3, 4), 8)

